### PR TITLE
Why Node.js is good for online store

### PR DESCRIPTION
**Pull request by verah-tech**
Article submitted

### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
This fixes #1726

File: articles/why-nodejs-good-for-online-store/index.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+**The benefit of the plugins and packages** - Node.js comes with many packages and plugins applied to most apps. That way, the developers do not need to write a program from scratch but can use open source solutions. It significantly reduces the time of development. It also comprises packages for e-commerce.
+
+**Frontend and Backend Language** - Popular frontend frameworks such as [React](https://reactjs.org/tutorial/tutorial.html), [Ember](https://guides.emberjs.com/release/tutorial/part-1/), and [Angular](https://angular.io/tutorial) use Node.js language. Node.js language can create applications that can run on the frontend and backend. The development process is made easy because there is clear communication between front and back end teams since they understand the two [codebases](https://www.techopedia.com/definition/23962/codebase) better. Also, the development team can be smaller and more efficient. In scaling up, recruiting new team members with no much technological diversity will be easier.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:14]: Node.js is not a language. Could you correct it.

File: articles/why-nodejs-good-for-online-store/index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+
+### Introduction
+[Node.js](https://nodejs.dev/learn) is a JavaScript environment used to develop high-performance applications that are scalable and handles multiple I/O operations. E-commerce systems involve several modules working together. Each module, with its complexities, makes Node.js perfect for developing such a system.
+### Advantages of building with Node.js
+
+**Performance** - Node.js gives better results with comparatively lower costs as it supports multitasking. It is convenient compared to other languages since an e-commerce store involves several operations such as adding items to the cart, changing the item features, choosing preferred payment options. From the performance point of view, the technology to be used has to serve such tasks efficiently, and Node.js does this effectively due to its ability to handle different operations simultaneously, thus making it a good choice for an online store.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:6]: `It is convenient compared to other languages since an e-commerce store`
You compared Node.js to languages.
Node.js is not a language. Could you correct it.

File: articles/why-nodejs-good-for-online-store/index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+**Real-Time Communication and Easy Management** – Real-time interaction with the customer for an e-commerce store is essential. Since Node.js supports frontend and backend development, it is practical to implement a chat-based platform/tool, whereby the interaction with clients will be instant and communication made easy. With the use of the [NoSQL](https://www.mongodb.com/nosql-explained) database that Node.js supports, the management of product listing is simplified, and the e-commerce admin can find it convenient to manage products whether from it is from a single vendor or multivendor platform.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:16]: and the e-commerce admin can find it convenient to manage products` whether from it is from a `single vendor or multivendor platform.

Could you correct this sentence.

File: articles/why-nodejs-good-for-online-store/index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+**Data Streaming** – With Node.js, HTTP requests and responses are processed as continuous data streams instead of considered isolated events. It ensures better processing and faster load times. It is essential for e-commerce since data streaming ensures faster uploads and hence better interactions.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:18]: With Node.js, HTTP requests and responses are processed as continuous data streams instead of `considered` isolated events

Remove considered.

File: articles/why-nodejs-good-for-online-store/index.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+**Non-Blocking Code Execution –** Node.js platform is built to handle asynchronous I/O and manage the process without any delay, which means it does away with the possibility of any delay in execution. It is not the case compared to synchronous code execution where execution is sequential, meaning one thread cannot execute unless the previous one is finished. It would bring delay in overall execution and hence more response time. In an e-commerce store, each customer request should be handled simultaneously instead of sequentially as many clients are accessing the store simultaneously. Node.js handles this correctly since it allows simultaneous processing.
+
+### Platforms that use Node.js to build an E-Commerce Store
+Here are the few ready solutions that create E-commerce app:
+
+**Prime Fusion** – This is based on a JavaScript framework that includes MongoDB, Express, Angular.js, Node.js, GraphQL, webpack React, Gulp, Babel Mocha. The databases and frameworks mentioned above provide the developers with the required tools to build an online store. Prime Fusion does not require special knowledge about building a store since it is backed with the framework.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:25]: Could you rephrase this `Prime Fusion does not require special knowledge about building a store since it is backed with the framework.` it seems difficult to understand.

File: articles/why-nodejs-good-for-online-store/index.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+**Reaction Commerce** – This platform is built on the Meteor framework, a fast, flexible, and scalable solution that works well with npm, Docker, and React. The platform codebase is modular, meaning that it can be customized to meet the developer's needs. The platform provides real-time analytics to assist the users in making better business decisions.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:27]: change `The platform codebase ...` to `The platform's codebase...`

File: articles/why-nodejs-good-for-online-store/index.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+### Platforms that use Node.js to build an E-Commerce Store
+Here are the few ready solutions that create E-commerce app:
+
+**Prime Fusion** – This is based on a JavaScript framework that includes MongoDB, Express, Angular.js, Node.js, GraphQL, webpack React, Gulp, Babel Mocha. The databases and frameworks mentioned above provide the developers with the required tools to build an online store. Prime Fusion does not require special knowledge about building a store since it is backed with the framework.
+
+**Reaction Commerce** – This platform is built on the Meteor framework, a fast, flexible, and scalable solution that works well with npm, Docker, and React. The platform codebase is modular, meaning that it can be customized to meet the developer's needs. The platform provides real-time analytics to assist the users in making better business decisions. 
+
+**Trader io** – This is also built on the MEAN stack. It offers flexible and scalable API and has basic functionalities e-commerce store needs. The developers had previously created several e-commerce applications, and they took the existing code and made it open source.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:29]: Could you explain what the `MEAN stack` entails for beginners to understand.

File: articles/why-nodejs-good-for-online-store/index.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+### What to consider on whether to use Node.js
+A few factors can be considered on whether to use Node.js as a preferred development solution. Some of them include:
+
+**Choosing the right solution** - They are ready-made solutions that use Node.js to choose from, as mentioned above. It depends on the user's needs, the number of features the system should have, and its complexity. Also, one can build the app from scratch depending on the unique feature the app is intended to have that ready-made solutions are missing.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:34]: `They` are ready-made solutions that use Node.js to choose from, as mentioned above.  

There are ......

File: articles/why-nodejs-good-for-online-store/index.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+**Technology Coherence** - Node.js only serves as a backend for an e-commerce store; other technologies can be considered when building other application parts to avoid any problems that may appear when developing the app. It is better to use a ready package offered by e-commerce platforms, as earlier discussed, to ensure that all the frameworks and solutions remain consistent.
```
Comment (WanjaMIKE) [inline articles/why-nodejs-good-for-online-store/index.md:36]: In this sentence `It is better to use a ready package offered by e-commerce platforms ...`  replace `ready` with `existing`

File: authors/verah-ombui/index.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
+---
+title: Verah Ombui
+type: engineering-education/author
+---
+Verah Ombui is a undergraduate Computer Science student at Jomo Kenyatta University Of Agriculture and Technology. His interests are web development, cloud computing, and data analysis science. Verah is  a Technology enthusiast.
```
Comment (WanjaMIKE) [inline authors/verah-ombui/index.md:5]: Verah Ombui is `an` ...  also is Verah a girl `His interests are web development,`

**Review by WanjaMIKE — CHANGES_REQUESTED** [review]
@verah-tech Hey, I have really learned a lot about Node.js and its benefits. Could you please implement the above minor suggestions before we proceed to the next level.

**Comment by verah-tech** [comment]
The changes has been implemented

**Review by WanjaMIKE — APPROVED** [review]
@verah-tech Kindly, note that you will need to work more on grammar. Avoid long sentences and repetitions to improve quality.

**Comment by verah-tech** [comment]
@WanjaMIKE point noted future writings will be better.

**Comment by hectorkambow** [comment]
plagiarism check done
